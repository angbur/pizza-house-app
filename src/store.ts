import { configureStore } from '@reduxjs/toolkit';
import orderReducer from 'components/Pages/Order/orderSlice';
import { pizzaApi } from 'services/pizza';

export const store = configureStore({
  reducer: {
    order: orderReducer,
    [pizzaApi.reducerPath]: pizzaApi.reducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(pizzaApi.middleware),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
