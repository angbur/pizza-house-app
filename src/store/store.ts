import { configureStore, Store } from '@reduxjs/toolkit';
import orderReducer from 'components/Pages/Order/orderSlice';
import dialogReducer from 'components/Organisms/Dialog/dialogSlice';
import { pizzaApi } from 'services/pizza';
import orderMiddleware from './middleware/orderMiddleware';

export const store: Store = configureStore({
  reducer: {
    order: orderReducer,
    dialog: dialogReducer,
    [pizzaApi.reducerPath]: pizzaApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(pizzaApi.middleware, orderMiddleware),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
