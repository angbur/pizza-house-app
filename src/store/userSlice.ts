import { createSlice } from '@reduxjs/toolkit';
import { RootState } from './store';
import { toast } from 'react-toastify';
import { userApi } from 'services/user';

type User = {
  loggedIn: boolean;
  status: string | undefined;
  error: string | undefined;
  userId: string | undefined;
  login: string | undefined;
  token: string | undefined;
};

const initialState: User = {
  loggedIn: false,
  status: 'idle',
  error: undefined,
  userId: undefined,
  login: undefined,
  token: undefined,
};

const userSlice = createSlice({
  name: 'user',
  initialState,
  reducers: {
    logout: (state) => {
      state.loggedIn = false;
      state.userId = undefined;
      state.login = undefined;
      state.token = undefined;
      toast.success('Logged out!');
    },
  },
  extraReducers: (builder) => {
    builder.addMatcher(userApi.endpoints.login.matchFulfilled, (state, { payload }) => {
      state.loggedIn = true;
      state.token = payload.token;
      state.userId = payload.userId;
      state.login = payload.login;
    });
  },
});

export const { logout } = userSlice.actions;

export const selectName = (state: RootState) => state.user.user;

export const isLoggedIn = (state: RootState) => state.user.loggedIn;

export const selectResponseStatus = (state: RootState) => state.user.status;

export const selectResponseError = (state: RootState) => state.user.error;

export const selectLoggedInUser = (state: RootState) => state.user.user;

export default userSlice.reducer;
