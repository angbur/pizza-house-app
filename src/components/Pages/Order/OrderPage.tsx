import Button from 'components/Atoms/Button/Button';
import Typography from 'components/Atoms/Typography/Typography';
import { openDialog, FormType } from 'components/Organisms/Dialog/dialogSlice';
import { useAppDispatch, useAppSelector } from 'hooks';
import { useNavigate } from 'react-router-dom';
import { selectIsLoggedIn } from 'store/userSlice';
import styled from 'styled-components';
import OrderList from './components/OrderList/OrderList';
import { removeAllOrder, selectListOfOrderItems } from './orderSlice';

const Container = styled.section`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  text-align: center;
  width: 100%;
  & > h2 {
    margin: 2rem 0;
  }
`;

const StyledDiv = styled.div`
  display: flex;
  gap: 3rem;
  margin: 2rem;
`;

const OrderPage = () => {
  const navigate = useNavigate();
  const dispatch = useAppDispatch();
  const isLoggedIn = useAppSelector(selectIsLoggedIn);
  const orderList = useAppSelector(selectListOfOrderItems);

  const handleRemoveAll = () => {
    dispatch(removeAllOrder());
  };

  const handleNavigate = () => {
    if (!isLoggedIn) dispatch(openDialog(FormType.login));
    navigate('/order/form');
  };
  return (
    <Container>
      <Typography variant='section-title' color='light'>
        My Order
      </Typography>
      <OrderList />
      {orderList.length !== 0 ? (
        <StyledDiv>
          <Button variant='button-text-light' onClick={handleRemoveAll}>
            Remove all
          </Button>
          <Button variant='primary-light' onClick={handleNavigate}>
            Go to order form
          </Button>
        </StyledDiv>
      ) : null}
    </Container>
  );
};

export default OrderPage;
