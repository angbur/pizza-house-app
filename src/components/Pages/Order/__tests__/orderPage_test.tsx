import renderer from 'react-test-renderer';
import ThemeProvider from 'components/Theme/ThemeContext';
import { store } from 'store/store';
import { Provider } from 'react-redux';
import OrderPage from '../OrderPage';
import { BrowserRouter } from 'react-router-dom';

const Wrapper = () => {
  return (
    <ThemeProvider>
      <BrowserRouter>
        <Provider store={store}>
          <OrderPage />
        </Provider>
      </BrowserRouter>
    </ThemeProvider>
  );
};

describe('Order Page', () => {
  test('should renders order page correctly', () => {
    const tree = renderer.create(<Wrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
