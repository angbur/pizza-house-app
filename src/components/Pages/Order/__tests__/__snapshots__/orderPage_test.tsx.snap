// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Page should renders order page correctly 1`] = `
<section
  className="sc-bjfHbI daFCJp"
>
  <h2
    className="section-title"
    style={
      Object {
        "color": "#FCFBFB",
        "fontFamily": "Source Sans Pro",
        "fontSize": "2.25rem",
        "fontWeight": "300",
        "letterSpacing": "1px",
        "lineHeight": "1.3",
      }
    }
  >
    My Order
  </h2>
  <p
    className="paragraph"
    style={
      Object {
        "color": "#FCFBFB",
        "fontFamily": "Source Sans Pro",
        "fontSize": "1rem",
        "fontWeight": "400",
        "letterSpacing": "0",
        "lineHeight": "1.6",
      }
    }
  >
    You have not ordered anything yet.
  </p>
</section>
`;
